### DISCUSSION TITLE
AppwriteException: Network request failed

### DISCUSSION THREAD

**Discussion by devrudraa** [appwrite/appwrite#5114] [Q&A] [answered]
### 👟 Reproduction steps

When I try to connect with the database it gave me the error [AppwriteException: Network request failed]

```javascript
    import { Client, Databases } from "appwrite";
    const client = new Client();

    client
        .setEndpoint('http://localhost/v1')
        .setProject('63e740a254762a08ca35');

    const databases = new Databases(client);
    const promise = databases.listDocuments('63e76696d2afb0a53a72', '63e766a12cbc17556296');

    promise.then(function (response) {
        console.log(response); // Success
    }, function (error) {
        console.log(error); // Failure
    });
```

### 👍 Expected behavior

It should give back me the list of documents of that database

### 👎 Actual Behavior

It is giving am Error: [AppwriteException: Network request failed]

### 🎲 Appwrite version

Version 1.2.x

### 💻 Operating system

Linux

### 🧱 Your Environment

I am using Docker(Docker version 20.10.12, build 20.10.12-0ubuntu4) for this and then I installed appwrite by following the docs and configured it to the **localhost**

and for react native I am using Android Studio Android Emulator Pixel 5 phone.
and for the internet connection, I connect my phone to the PC and switched on the **USB Threading** my react app works fine on the emulator but for the appwrite I am getting the error.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
@devrudraa, thanks for creating this issue! 🙏 The problem might be the usage of localhost in a mobile application. Localhost refers to the current host so your app might be trying to connect to the Android device rather than Appwrite. If your two devices are in the same network, I would suggest using your PC's LAN IP instead.of localhost.

**Comment by devrudraa** [comment]
I did that I am glad that not it is giving a new error :)
**[AppwriteException: The current user is not authorized to perform the requested action.]**
i used my **IPv4** address 
**.setEndpoint('http://IPv4 Address/v1')**
and now it gives another error.

**Reply by stnguyen90** [reply]
@devrudraa, let's not re-use the issue for other problems. 

That said, you see that error because you're trying to make an API call that the user is not allowed to make.

I encourage you to join us on [Discord](https://appwrite.io/discord/) if you have additional questions.
